cub/cub/device/device_merge.cuh

```diff
@@ -0,0 +1,197 @@
+
+#include <cub/config.cuh>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cub/detail/nvtx.cuh>
+#include <cub/device/dispatch/dispatch_merge.cuh>
+#include <cub/util_namespace.cuh>
+
+#include <cuda/std/functional>
+
+CUB_NAMESPACE_BEGIN
+
+//! DeviceMerge provides device-wide, parallel operations for merging two sorted sequences of values (called keys) or
+//! key-value pairs in device-accessible memory. The sorting order is determined by a comparison functor (default:
+//! less-than), which has to establish a [strict weak ordering].
+//!
+//! [strict weak ordering]: https://en.cppreference.com/w/cpp/concepts/strict_weak_order
+struct DeviceMerge
+{
+  //! @rst
+  //! Overview
+  //! +++++++++++++++++++++++++++++++++++++++++++++
+  //! Merges two sorted sequences of values (called keys) into a sorted output sequence. Merging is unstable,
+  //! which means any two equivalent values (neither value is ordered before the other) may be written to the ouput
+  //! sequence in any order.
+  //!
+  //! A Simple Example
+  //! +++++++++++++++++++++++++++++++++++++++++++++
+  //! The code snippet below illustrates the merging of two device vectors of `int` keys.
+  //!
+  //! .. literalinclude:: ../../../cub/test/catch2_test_device_merge_api.cu
+  //!     :language: c++
+  //!     :dedent:
+  //!     :start-after: example-begin merge-keys
+  //!     :end-before: example-end merge-keys
+  //!
+  //! @endrst
+  //!
+  //! @tparam KeyIteratorIn1 **[deduced]** Random access iterator to the first sorted input sequence. Must have the same
+  //! value type as KeyIteratorIn2.
+  //! @tparam KeyIteratorIn2 **[deduced]** Random access iterator to the second sorted input sequence. Must have the
+  //! same value type as KeyIteratorIn1.
+  //! @tparam KeyIteratorOut **[deduced]** Random access iterator to the output sequence.
+  //! @tparam CompareOp **[deduced]** Binary predicate to compare the input iterator's value types. Must have a
+  //! signature equivalent to `bool operator()(Key lhs, Key rhs)` and establish a [strict weak ordering].
+  //!
+  //! @param[in] d_temp_storage Device-accessible allocation of temporary storage. When `nullptr`, the required
+  //! allocation size is written to `temp_storage_bytes` and no work is done.
+  //! @param[in,out] temp_storage_bytes Reference to size in bytes of `d_temp_storage` allocation.
+  //! @param[in] keys_in1 Iterator to the beginning of the first sorted input sequence.
+  //! @param[in] num_keys1 Number of keys in the first input sequence.
+  //! @param[in] keys_in2 Iterator to the beginning of the second sorted input sequence.
+  //! @param[in] num_keys2 Number of keys in the second input sequence.
+  //! @param[in] keys_out Iterator to the beginning of the output sequence.
```

Comment:
suggestion: 
```suggestion
  //! @param[out] keys_out Iterator to the beginning of the output sequence.
```

Reply:
:D The output iterator is still an input parameter, because it is copied into the function. Out is just written to storage reachable via the input parameter.

But I can mark it as you want.